scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
     else
-        echo "Installation failed: Could not find dpkg or rpm on the system"
-        exit 1
+      if [ -z "$FAILED_PREREQS" ]; then
+        FAILED_PREREQS="${fg_red}$prerequisite${reset}"
+      else
+        FAILED_PREREQS="$FAILED_PREREQS, ${fg_red}$prerequisite${reset}"
+      fi
     fi
+  done
+
+  if [ -n "$FAILED_PREREQS" ]; then
+    failed
+    error_exit "The following dependencies are required by this script: [$FAILED_PREREQS]"
+  fi
+  succeeded
+  return 0
 }
 
-set_file_name() {
-    package_file_name="observiq-otel-collector_linux_${arch}.${package_type}"
-    out_file_path="$TMP_DIR/$package_file_name"
+# This will check to ensure either dpkg or rpm is installedon the system
+package_type_check()
+{
+  info "Checking for package manager..."
+  if command -v dpkg > /dev/null 2>&1; then
+      succeeded
+      pkg_type="deb"
+      return 0
+  elif command -v rpm > /dev/null 2>&1; then
+      succeeded
+      pkg_type="rpm"
+      return 0
+  else
+      failed
+      error_exit "Could not find dpkg or rpm on the system"
+  fi
 }
 
-download_package() {
-    download_url="${DOWNLOAD_URL_BASE}/latest/download/${package_file_name}"
-    
-    echo "Downloading ${download_url} to ${out_file_path}..."
-    curl -L "$download_url" -o "$out_file_path" --progress-bar --fail
-    echo "${out_file_path} successfully downloaded!"
-}
-
-package_version() {
-    case "$package_type" in
-        deb)
-            dpkg -s observiq-otel-collector | grep -i '^Version:' | cut -d' ' -f2
-            ;;
-        rpm)
-            rpm -q --queryformat '%{VERSION}' "observiq-otel-collector"
-            ;;
-        *)
-            echo "unknown"
-            ;;
-    esac
-}
-
-install_package() {
-    case "$package_type" in
-        deb)
-            dpkg -i "$out_file_path"
-            ;;
-        rpm)
-            # -U installs OR upgrades
-            # just -i will fail if previous versions exists
-            rpm -U "$out_file_path"
-            ;;
-        *)
-            echo "Installation failed: Could not determine the package manager to use for installation (package_type: ${package_type})"
-            exit 1
-            ;;
-    esac
-
-    if command -v systemctl > /dev/null 2>&1; then
-        systemctl enable --now observiq-otel-collector
+# This will install the package by downloading the archived collector,
+# extracting the binaries, and then removing the archive.
+install_package()
+{
+  banner "Installing observIQ OpenTelemetry Collector"
+  increase_indent
+
+  proxy_args=""
+  if [ -n "$proxy" ]; then
+    proxy_args="-x $proxy"
+    if [ -n "$proxy_user" ]; then
+      proxy_args="$proxy_args -U $proxy_user:$proxy_password"
     fi
+  fi
+
+  if [ -n "$proxy" ]; then
+    info "Downloading package using proxy..."
+  fi 
+
+  info "Downloading package..."
+  eval curl -L "$proxy_args" "$collector_download_url" -o "$out_file_path" --progress-bar --fail || error_exit "$LINENO" "Failed to download package"
```

Comment:
I think this is correct, but the eval is confusing me so I just want to verify; "$proxy_args" DOES split here, right? Because of the eval?

Follow up question, would that cause "out_file_path" to split if it had spaces in it?

Reply:
I'm not sure to be honest. 

I don't know what happens to `out_file_path` if it has spaces. I think that's unlikely (not impossible). Most linux admins are not going to have a directory path with spaces in it. I can test it out though to see what happens.

Reply:
So it doesn't work with spaces in the path. Our current stanza and BP scripts will have this same issue since I copied this line. I'm ok accepting this risk as spaces in directory names is not common in linux.

@jsirianni @BinaryFissionGames are you also ok with this restriction?